npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb

```diff
@@ -25,6 +25,14 @@ def self.yarn_version_numeric(yarn_lock)
         end
       end
 
+      def self.pnpm_version_numeric(pnpm_lock)
+        if pnpm_lockfile_version(pnpm_lock).to_f >= 5.4
+          8
+        else
```

Comment:
This is a bit strange, in particular the `>= 5.4` translation to 8

Could you please add some inline comment to explain this?

Reply:
Thank you for the explanation! 

In terms of what to do, I think that if we do not support an specific lockfile format, we should fail fast. We should not cause damage or receive reports for unsupported versions

Reply:
After thinking about this a bit more, I think this is a strict improvement over what we have now.

This brings PNPM in line with what we already do for NPM and Yarn in terms of guessing the version from the lockfile format, and should transform unknown errors due to incompatibilities into user errors (even if the error message is not ideal).

If the new error message comes as confusing when a specific PNPM version is not explicitly requested in `package.json`, we can iterate and give a more specific error that points to the lockfile format instead, but I don't want to block this PR on that.